### PR TITLE
Only display the modified post date if the post has been modified

### LINKED ISSUE #46645: Post date block: to display post date and modified date side by side.
I want to display the post date and modified date side by side.
To do this, I need the option of not displaying the modified date on the front if it has never been modified.
I think we also need to add a class to distinguish each of them.

### PR DESCRIPTION
**Pull request by carolinan**
## What?

The post date block has a setting that displays the modified date. When the setting is enabled, it displays the post date
if the post has not been modified.

This PR updates the block to only display the modified date if the setting is enabled and the post has been modified.
If the setting is enabled, but the post has not been modified, nothing is shown on the front.
It also adds a new CSS class to identify that the setting is enabled so that developers can style it differently.

Fixes [#46645](https://github.com/WordPress/gutenberg/issues/46645)

## Why?

Without this PR, a template that displays both the published date and the modified date could show two identical dates,
and there was no way to hide the modified date if the post was not modified.

## How?

The PR compares the modified date and the post date and returns early if the time difference is a less than a minute.

## Testing Instructions

1. Create a new post.
2. Add a post date block.
3. _Wait a minute or two_ and add a second post date block. Enable the "Display last modified date" setting.
4. Save and view the front. Both blocks should show, with different dates.

5. Create a second new post.
6. Add a post date block. Enable the "Display last modified date" setting.
7. Save and view the front. If you were fast enough, the block should not show.
8. Go back to the editor, _wait a minute_ and make a change to the post.
9. Save and view the front, confirm that the block shows.

### CODE REVIEW

**Review by aristath — APPROVED** [review]
Thank you for working on this one @carolinan, this is a nice change!
Code looks good to me, and it does what it says. This looks OK to merge 👍

File: packages/block-library/src/post-date/index.php [hunk 1]
```diff
@@ -18,25 +18,34 @@ function render_block_core_post_date( $attributes, $content, $block ) {
...
+	 * If the "Display last modified date" setting is enabled,
+	 * only display the modified date if it is later than the publishing date.
+	 */
 	if ( isset( $attributes['displayType'] ) && 'modified' === $attributes['displayType'] ) {
-		$formatted_date   = get_the_modified_date( empty( $attributes['format'] ) ? '' : $attributes['format'], $post_ID );
-		$unformatted_date = esc_attr( get_the_modified_date( 'c', $post_ID ) );
-	} else {
-		$formatted_date   = get_the_date( empty( $attributes['format'] ) ? '' : $attributes['format'], $post_ID );
-		$unformatted_date = esc_attr( get_the_date( 'c', $post_ID ) );
+		if ( get_the_modified_date( 'Y-m-d h:i', $post_ID ) > get_the_date( 'Y-m-d h:i', $post_ID ) ) {
```
Comment (aristath) [inline packages/block-library/src/post-date/index.php]: In order for the `>` comparison to work, perhaps we should format the string like this so it's a number?

```suggestion
		if ( get_the_modified_date( 'Ymdhi', $post_ID ) > get_the_date( 'Ymdhi', $post_ID ) ) {
```

**Comment by bugnumber9** [comment]
This change made it impossible to use `<!-- wp:post-date {"displayType":"modified"} /-->` and expect the consistent behavior.
Really, if someone wants to display both post date and modified date and wants to avoid both displaying the same date - that's when you'd apply some custom logic to avoid this.
IMO, `<!-- wp:post-date {"displayType":"modified"} /-->` should never return an empty value, just as `get_the_modified_date` doesn't.
Ref: https://github.com/WordPress/gutenberg/issues/57915

**Comment by burtrw** [comment]
Wanted to leave feedback that this change caught me out on a new site this week. I only want to show one date on the post, but if the post is never updated, I don't see any date now. I also wanted to use the row block to add text before the date, but I can't customize it if it is published/updated.

image

Conditional display plugins can help solve this, but it would be ideal to have a solution in core since it is likely widely used, especially with SEO trends towards displaying the lastest update date.

**Pull request merged**